# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/mac/MOXTextMarkerDelegate.h
# Commit: bb7606e92697
# Full Hash: bb7606e92697cbf94bd0b7fe0710acd53041cfa5
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-11-21 21:37:46
# Regressor Bug: 1672700
# File Overlap Count: 1
# Description:
#   Bug 1672700 - Make mozTextAccessible attribute getters use GeckoTextMarker. r=morgan
#   
#   This allows `contenteditable` textboxes to work correctly with flattened text values.
#   
#   The attribute setters, aside from AXValue, don't work in Chrome or Safari with aria textboxes,
# ==============================================================================

diff -r 1ad690f5f41a -r bb7606e92697 accessible/mac/MOXTextMarkerDelegate.h
--- a/accessible/mac/MOXTextMarkerDelegate.h	Sat Nov 21 04:13:43 2020 +0000
+++ b/accessible/mac/MOXTextMarkerDelegate.h	Sat Nov 21 04:13:45 2020 +0000
@@ -8,6 +8,7 @@
 #import <Cocoa/Cocoa.h>
 
 #import "MOXAccessibleProtocol.h"
+#import "GeckoTextMarker.h"
 
 #include "AccessibleOrProxy.h"
 
@@ -31,6 +32,8 @@
 
 - (void)invalidateSelection;
 
+- (mozilla::a11y::GeckoTextMarkerRange)selection;
+
 // override
 - (id)moxStartTextMarker;
 